src/dmd/doc.d

```diff
@@ -1815,12 +1814,12 @@ extern (C++) int cmp(const(char)* stringz, const(void)* s, size_t slen)
     return memcmp(stringz, s, slen);
 }
 
-extern (C++) int icmp(const(char)* stringz, const(void)* s, size_t slen)
+private bool icmp(const(char)* stringz, const(char)* s, size_t slen)
 {
     size_t len1 = strlen(stringz);
```

Comment:
Your fixes make all the first arguments to `icmp` a string, but then they get coerced to pointers by the parameter type, and then strlen is run on them here. Just commit to having string parameters to icmp.